### DISCUSSION TITLE
Assembly Workbench needs new important features, yet!

### DISCUSSION THREAD

**Discussion by TheAliKing** [FreeCAD/FreeCAD#18215] [Discussions that will turn in to issues but need refining]
Hi
I think that Assembly Workbench still needs four new features:
1. Tangent Constraint (Closed Loop Path like a Cam Mechanism)
2. Tangent Constraint (Along Open Path)
3. Symmetric Constraint (making two things symmetric to each other based on centering reference such as a Plane, Face, Line, or Point)
4. Mirroring Assembly (Right now, FreeCad Doesn't have any option for this)
Among all of these features the Symmetric Constraint is the most essential and important one.

**Comment by luzpaz** [comment]
Please make sure these features aren't already reported in the bugtracker. If they aren't please open a feature request for each. These meta-feature discussions that list multiple features in one discussion become complex really fast due to limitations in how to discuss nuances of features.

**Reply by TheAliKing** [reply]
ok, you're right. I'm new to GitHub workflow, and I just wanted to suggest what assembly workbench is missing. so, I will check it anybody 
has reported these features and report the one that nobody did.
Anyway, Thanks for your advice.
